File: cpp/Settings/BaseSetting.c
// Type: Settings.BaseSetting
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Settings/BaseSetting.cs
// Prior real C# source (older reference): Assets/Scripts/Settings/SettingTypes/BaseSetting.cs
// --------------------------------

// Settings.BaseSetting$$SerializeToJsonString
// il2cpp: System_String_o* Settings_BaseSetting__SerializeToJsonString (Settings_BaseSetting_o* __this, const MethodInfo* method);
// 0x41fddc0

System_String_o *
Settings_BaseSetting__SerializeToJsonString(Settings_BaseSetting_o *__this,MethodInfo *method)

{
  long lVar1;
  Settings_BaseSetting_c *pSVar2;
  Il2CppMethodPointer vtableDispatch;
  long *plVar3;
  System_String_o *pSVar4;
  SimpleJSONFixed_JSONNode_o *pSVar5;
  MethodInfo *aJSON;
  
  aJSON = (__this->klass->vtable)._5_unknown.method;
  plVar3 = (long *)(*(__this->klass->vtable)._5_unknown.methodPtr)();
  if (plVar3 != (long *)0x0) {
    lVar1 = *plVar3;
    pSVar4 = (System_String_o *)
             (**(code **)(lVar1 + 0x318))
                       (plVar3,4,*(undefined8 *)(lVar1 + 800),lVar1,*(code **)(lVar1 + 0x318));
    return pSVar4;
  }
  il2cpp_runtime_helper_022b2c90();
  pSVar5 = SimpleJSONFixed_JSON__Parse((System_String_o *)aJSON,(MethodInfo *)0x0);
  pSVar2 = __this->klass;
  vtableDispatch = (pSVar2->vtable)._6_unknown.methodPtr;
  pSVar4 = (System_String_o *)
           (*vtableDispatch)
                     (__this,pSVar5,(pSVar2->vtable)._6_unknown.method,pSVar2,vtableDispatch);
  return pSVar4;
}


// Settings.BaseSetting$$DeserializeFromJsonString
// il2cpp: void Settings_BaseSetting__DeserializeFromJsonString (Settings_BaseSetting_o* __this, System_String_o* json, const MethodInfo* method);
// 0x41fde00

void Settings_BaseSetting__DeserializeFromJsonString
               (Settings_BaseSetting_o *__this,System_String_o *json,MethodInfo *method)

{
  Settings_BaseSetting_c *pSVar1;
  Il2CppMethodPointer vtableDispatch;
  SimpleJSONFixed_JSONNode_o *pSVar2;
  
  pSVar2 = SimpleJSONFixed_JSON__Parse(json,(MethodInfo *)0x0);
  pSVar1 = __this->klass;
  vtableDispatch = (pSVar1->vtable)._6_unknown.methodPtr;
  (*vtableDispatch)(__this,pSVar2,(pSVar1->vtable)._6_unknown.method,pSVar1,vtableDispatch);
  return;
}


// Settings.BaseSetting$$Copy
// il2cpp: void Settings_BaseSetting__Copy (Settings_BaseSetting_o* __this, Settings_BaseSetting_o* other, const MethodInfo* method);
// 0x41fde30

void Settings_BaseSetting__Copy
               (Settings_BaseSetting_o *__this,Settings_BaseSetting_o *other,MethodInfo *method)

{
  Settings_BaseSetting_c *pSVar1;
  Il2CppMethodPointer vtableDispatch;
  undefined8 uVar2;
  
  if (other != (Settings_BaseSetting_o *)0x0) {
    uVar2 = (*(other->klass->vtable)._5_unknown.methodPtr)(other,(other->klass->vtable)._5_unknown.method);
    pSVar1 = __this->klass;
    vtableDispatch = (pSVar1->vtable)._6_unknown.methodPtr;
    (*vtableDispatch)(__this,uVar2,(pSVar1->vtable)._6_unknown.method,pSVar1,vtableDispatch);
    return;
  }
  il2cpp_runtime_helper_022b2c90();
  System_Object___ctor((Il2CppObject *)__this,(MethodInfo *)0x0);
  return;
}


// Settings.BaseSetting$$.ctor
// il2cpp: void Settings_BaseSetting___ctor (Settings_BaseSetting_o* __this, const MethodInfo* method);
// 0x41fde70

void Settings_BaseSetting___ctor(Settings_BaseSetting_o *__this,MethodInfo *method)

{
  System_Object___ctor((Il2CppObject *)__this,(MethodInfo *)0x0);
  return;
}

File: csharp_body_deref/Scripts/Settings/BaseSetting.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Settings.BaseSetting
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Settings/BaseSetting.c
// Prior real C# source (older reference): Assets/Scripts/Settings/SettingTypes/BaseSetting.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;
using SimpleJSONFixed;

namespace Settings;

[Token(Token = "0x20000AD")]
internal abstract class BaseSetting
{
	[Token(Token = "0x600036A")]
	public abstract void SetDefault();

	[Token(Token = "0x600036B")]
	public abstract JSONNode SerializeToJsonObject();

	[Token(Token = "0x600036C")]
	public abstract void DeserializeFromJsonObject(JSONNode json);

	[Token(Token = "0x600036D")]
	[Address(RVA = "0x40FDDC0", Offset = "0x40FDDC0", VA = "0x40FDDC0", Slot = "7")]
	public virtual string SerializeToJsonString()
	{
		return null;
	}

	[Token(Token = "0x600036E")]
	[Address(RVA = "0x40FDE00", Offset = "0x40FDE00", VA = "0x40FDE00", Slot = "8")]
	public virtual void DeserializeFromJsonString(string json)
	{
	}

	[Token(Token = "0x600036F")]
	[Address(RVA = "0x40FDE30", Offset = "0x40FDE30", VA = "0x40FDE30", Slot = "9")]
	public virtual void Copy(BaseSetting other)
	{
	}

	[Token(Token = "0x6000370")]
	[Address(RVA = "0x40FDE70", Offset = "0x40FDE70", VA = "0x40FDE70")]
	protected BaseSetting()
	{
	}
}
